src/Native/System.Security.Cryptography.Native/pal_ssl.cpp

```diff
@@ -42,6 +44,343 @@ extern "C" SSL_CTX* SslCtxCreate(SSL_METHOD* method)
     return SSL_CTX_new(method);
 }
 
+extern "C" int64_t SslCtxCtrl(SSL_CTX* ctx, int32_t cmd, int64_t larg, void* parg)
+{
+    return SSL_CTX_ctrl(ctx, cmd, larg, parg);
+}
+
+extern "C" SSL* SslCreate(SSL_CTX* ctx)
+{
+    return SSL_new(ctx);
+}
+
+extern "C" int32_t SslGetError(SSL* ssl, int32_t ret)
+{
+    return SSL_get_error(ssl, ret);
+}
+
+extern "C" void SslDestroy(SSL* ssl)
+{
+    if (ssl)
+    {
+        SSL_free(ssl);
+    }
+}
+
+extern "C" void SslCtxDestroy(SSL_CTX* ctx)
+{
+    if (ctx)
+    {
+        SSL_CTX_free(ctx);
+    }
+}
+
+extern "C" const char* SslGetVersion(SSL* ssl)
+{
+    return SSL_get_version(ssl);
+}
+
+/*
+The values used in OpenSSL for SSL_CIPHER algorithm_enc.
+*/
+enum class SSL_CipherAlgorithm : int64_t
+{
+#if HAVE_SSL_CIPHER_SPLIT_ALGORITHMS
+    SSL_DES = 1,
+    SSL_3DES = 2,
+    SSL_RC4 = 4,
+    SSL_RC2 = 8,
+    SSL_IDEA = 16,
+    SSL_eNULL = 32,
+    SSL_AES128 = 64,
+    SSL_AES256 = 128,
+    SSL_CAMELLIA128 = 256,
+    SSL_CAMELLIA256 = 512,
+    SSL_eGOST2814789CNT = 1024,
+    SSL_SEED = 2048,
+    SSL_AES128GCM = 4096,
+    SSL_AES256GCM = 8192
+#else
+    SSL_DES = 0x00008000L,
+    SSL_3DES = 0x00010000L,
+    SSL_RC4 = 0x00020000L,
+    SSL_RC2 = 0x00040000L,
+    SSL_IDEA = 0x00080000L,
+    //SSL_eFZA = 0x00100000L,  this value is defined in ssl_locl.h, but has no match
+    SSL_eNULL = 0x00200000L,
+    SSL_AES = 0x04000000L,
+    SSL_CAMELLIA = 0x08000000L,
+    SSL_SEED = 0x10000000L,
+#endif
+};
```

Comment:
See https://github.com/dotnet/corefx/pull/4026#discussion_r42600773

Honestly AFAIK, there isn't a way to check/guarantee these values since they aren't defined in headers on either my Linux or OSX machine. I sort of question whether what we are doing is a "public" API. The struct is visible, but it just exposes `unsigned long` members with no exposed defines for their values.